### PR TITLE
feat: refactor generation of snapshots from the cli

### PR DESCRIPTION
**Pull request by joshieDo**
* Refactors the generation of snapshots from the cli (dedup some code)
* Add parallelism if desired
* Add some stats that get printed at the end of the generation

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
smol nit,

pending @shekhirin

File: bin/reth/src/db/snapshots/mod.rs [hunk 1]
```diff
@@ -130,8 +148,102 @@ impl Command {
...
+        self.stats(created_snapshots?)
+    }
+
+    /// Prints detailed statistics for each snapshot, including loading time.
+    ///
+    /// This function loads each snapshot from the provided paths and prints
+    /// statistics about various aspects of each snapshot, such as filters size,
+    /// offset index size, offset list size, and loading time.
+    fn stats(&self, snapshots: Vec<impl AsRef<Path>>) -> eyre::Result<()> {
+        let mb = 1024.0 * 1024.0;
```
Comment (shekhirin) [inline bin/reth/src/db/snapshots/mod.rs:204]: nit: const

File: bin/reth/src/db/snapshots/mod.rs [hunk 2]
```diff
@@ -130,8 +148,102 @@ impl Command {
...
+        let avg_duration = total_duration / snapshots.len() as u32;
+
+        println!("Total Filters Size:     {:>7.2} MB", total_filters_size as f64 / mb);
+        println!("Total Offset Index Size: {:>7.2} MB", total_index_size as f64 / mb);
+        println!("Total Offset List Size:  {:>7.2} MB", total_offsets_size as f64 / mb);
+        println!("Total File Size:         {:>7.2} GB", total_file_size as f64 / (mb * 1024.0));
+        println!(
+            "Average Loading Time:    {:>7.2} ms | {:>7.2} µs",
+            avg_duration.as_millis() as f64,
+            avg_duration.as_micros() as f64
```
Comment (shekhirin) [inline bin/reth/src/db/snapshots/mod.rs:244]: can we just use the `Debug` impl for `Duration` here?

File: crates/storage/nippy-jar/src/filter/mod.rs [hunk 3]
```diff
@@ -36,4 +38,11 @@ impl InclusionFilter for InclusionFilters {
             InclusionFilters::Unused => todo!(),
         }
     }
+
+    fn size(&self) -> usize {
+        match self {
+            InclusionFilters::Cuckoo(c) => c.size(),
+            InclusionFilters::Unused => todo!(),
```
Comment (shekhirin) [inline crates/storage/nippy-jar/src/filter/mod.rs:45]: isn't it just `0` in that case?

**Review by shekhirin — APPROVED** [review]
LGTM, only nits

**Pull request merged**